src/DictVectors/initiators.jl

```diff
@@ -42,302 +78,147 @@ Concrete implementations:
 * [`Initiator`](@ref)
 * [`SimpleInitiator`](@ref)
 * [`CoherentInitiator`](@ref)
+* [`NonInitiator`](@ref)
+
+When defining a new `InitiatorRule`, also define the following:
+
+* [`initiator_valtype`](@ref)
+* [`from_initiator_value`](@ref)
+* [`to_initiator_value`](@ref)
+
+"""
+abstract type InitiatorRule end
+
+"""
+    initiator_valtype(rule::InitiatorRule, T)
 
-When defining a new `InitiatorRule`, also define a corresponding method for [`value`](@ref)!
+Return the `[`AbstractInitiatorValue{T}`](@ref) that is employed by the `rule`.
 """
-abstract type InitiatorRule{V} end
+initiator_valtype
 
 """
-    value(i::InitiatorRule, v::InitiatorValue)
-Convert the [`InitiatorValue`](@ref) `v` into a scalar value according to the
+    from_initiator_value(i::InitiatorRule, v::AbstractInitiatorValue)
+
+Convert the [`AbstractInitiatorValue`](@ref) `v` into a scalar value according to the
```

Comment:
Is the `InitiatorRule` argument really needed in order to convert the value into a scalar?

Reply:
The `InitiatorRule` holds the initiator threshold which tells us whether to keep or drop the value.